app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt

```diff
@@ -527,24 +524,19 @@ class NewMessageFragment : Fragment() {
         }
     }
 
-    override fun onStop() {
-        saveDraft()
-        super.onStop()
-    }
+    override fun onStop() = with(newMessageViewModel) {
 
-    private fun saveDraft() = with(newMessageViewModel) {
-        executeDraftActionWhenStopping(
-            action = if (shouldSendInsteadOfSave) DraftAction.SEND else DraftAction.SAVE,
-            isFinishing = requireActivity().isFinishing,
-            isTaskRoot = requireActivity().isTaskRoot,
-            subjectValue = binding.subjectTextField.text.toString(),
-            uiBodyValue = binding.bodyTextField.text.toString(),
-            startWorkerCallback = ::startWorker,
-        )
-    }
+        /**
+         * When the Activity is being stopped, we save the Draft.
+         * We then need the up-to-date subject & body values.
+         * If we are in the NewMessageFragment when stopping, it's easy, we just get them from the binding.
+         * If we are not (ex: FilePicker, IA…), we get them from the ViewModel.
```

Comment:
FilePicker is another Activity which means that from the point of view of "NewMessageActivity", the NewMessageFragment is still there and reachable